### PR TITLE
Structure

### PR DESCRIPTION
**Pull request by denpronin**

### CODE REVIEW

File: app/src/main/java/com/pronin/myphotomap/MapsActivity.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        dataGetter.getCameraImages(this, new DataGetter.OnGettingDataDoneListener() {
+            @Override
+            public void onGettingDataDone(List<Pictures> pictures) {
+                runOnUiThread(() -> {
+                    list.addAll(pictures);
+                });
+            }
+        });
+        // Add a marker in Sydney and move the camera
+        LatLng sydney = new LatLng(-34, 151);
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/MapsActivity.java]: Do you really need to do that?

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+import android.database.Cursor;
+import android.media.ExifInterface;
+import android.provider.MediaStore;
+import android.util.Log;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataGetter {
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: find better name for this class

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    public static final String CAMERA_IMAGE_BUCKET_NAME = "Camera";
+    private static final String TAG = "DataGetter";
+
+    public void getCameraImages(Context context, OnGettingDataDoneListener callback) {
+        Thread getting  = new Thread(() -> {
+                final String[] projection = {MediaStore.Images.Media.DATA, MediaStore.Images.Media.DATE_ADDED};
+                final String selection = MediaStore.Images.Media.BUCKET_DISPLAY_NAME + " = ?";// + " AND " + MediaStore.Images.Media.LONGITUDE + " != ?";
+                final String[] selectionArgs = {CAMERA_IMAGE_BUCKET_NAME};//, "null" };
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: spaces { ... }

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+                                Pictures pictures = new Pictures(data, imgLatLong[0], imgLatLong[1], cursor.getString(dateColumn));
+                                Log.d(TAG, pictures.toString());
+                                result.add(pictures);
+                            }
+                        } catch (IOException ex) {
+                            ex.printStackTrace();
+                        }
+                    } while (cursor.moveToNext());
+                } else {
+                    Log.d(TAG, "No images found!");
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: change log level to warning

File: app/src/main/java/com/pronin/myphotomap/model/Pictures.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
+package com.pronin.myphotomap.model;
+
+public class Pictures {
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/Pictures.java]: pictures -> picture.

File: app/src/main/java/com/pronin/myphotomap/model/Pictures.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        return longitude;
+    }
+
+    public String getPath() {
+        return path;
+    }
+
+    @Override
+    public String toString() {
+        return "Pictures{" +
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/Pictures.java]: don't forget to remove s in this place

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataGetter {
+    public static final String CAMERA_IMAGE_BUCKET_NAME = "Camera";
+    private static final String TAG = "DataGetter";
+
+    public void getCameraImages(Context context, OnGettingDataDoneListener callback) {
+        Thread getting  = new Thread(() -> {
+                final String[] projection = {MediaStore.Images.Media.DATA, MediaStore.Images.Media.DATE_ADDED};
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: { ... }

**Review by YahorHr — COMMENTED** [review]
not bad)

File: app/src/main/java/com/pronin/myphotomap/MapsActivity.java (hunk 1 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/MapsActivity.java]: yes

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java (hunk 2 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: renamed into PhotoArrayFormation

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java (hunk 3 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: ok

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java (hunk 4 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: ok

File: app/src/main/java/com/pronin/myphotomap/model/Pictures.java (hunk 5 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/model/Pictures.java]: renamed

File: app/src/main/java/com/pronin/myphotomap/model/Pictures.java (hunk 6 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/model/Pictures.java]: don't forgot

File: app/src/main/java/com/pronin/myphotomap/model/DataGetter.java (hunk 7 above)
Comment (denpronin) [reply app/src/main/java/com/pronin/myphotomap/model/DataGetter.java]: ok

File: app/src/main/java/com/pronin/myphotomap/model/LatLong.java [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+    private float longitude;
+
+    public LatLong() {}
+
+    public LatLong(float latArg, float longArg) {
+        latitude = new BigDecimal(latArg).setScale(2, RoundingMode.HALF_UP).floatValue();
+        longitude = new BigDecimal(longArg).setScale(2, RoundingMode.HALF_UP).floatValue();
+    }
+
+    public void setLatitude(float latitude) {
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/LatLong.java]: remove setters. you don't use them and you break logic you constructor

File: app/src/main/java/com/pronin/myphotomap/MapsActivity.java (hunk 1 above)
Comment (YahorHr) [reply app/src/main/java/com/pronin/myphotomap/MapsActivity.java]: why?

File: app/src/main/java/com/pronin/myphotomap/model/PhotoArrayFormation.java [hunk 9]
```diff
@@ -0,0 +1,68 @@
...
+        });
+        getting.start();
+
+    }
+
+    public interface OnGettingDataDoneListener {
+        void onGettingDataDone(List<Picture> pictures);
+    }
+
+    public HashMap<LatLong, ArrayList<Picture>> getMapMarkers(ArrayList<Picture> pictureArrayList) {
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/PhotoArrayFormation.java]: rename to createMarkersMap

File: app/src/main/java/com/pronin/myphotomap/model/LatLong.java [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+        this.latitude = latitude;
+    }
+
+    public void setLongitude(float longitude) {
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/LatLong.java]: remove

**Review by YahorHr — COMMENTED** [review]
commented

File: app/src/main/java/com/pronin/myphotomap/model/PhotoArrayFormation.java [hunk 11]
```diff
@@ -0,0 +1,86 @@
...
+                        }
+                    } while (cursor.moveToNext());
+                } else {
+                    Log.w(TAG, "No images found!");
+                }
+                cursor.close();
+                Log.d(TAG, Integer.toString(imgList.size()));
+                HashMap<LatLong, ArrayList<Picture>> markersMap = new HashMap<>(createMarkersMap(imgList));
+            if (!markersMap.isEmpty()) {
+                Log.d("!!!!!!!!!!!!!!!!!", "Map of markers size " + markersMap.size());
```
Comment (YahorHr) [inline app/src/main/java/com/pronin/myphotomap/model/PhotoArrayFormation.java]: change !!!!!!!!!! to class name

**Review by YahorHr — COMMENTED** [review]
merge to main

**Pull request merged**